packages/datadog-instrumentations/src/mariadb.js

```diff
@@ -0,0 +1,83 @@
+'use strict'
+
+const {
+  channel,
+  addHook,
+  AsyncResource
+} = require('./helpers/instrument')
+
+const shimmer = require('../../datadog-shimmer')
+
+const startCh = channel('apm:mariadb:query:start');
+const finishCh = channel('apm:mariadb:query:finish');
+const errorCh = channel('apm:mariadb:query:error');
+
+addHook(
+  {
+    name: 'mariadb',
+    file: 'lib/connection-promise.js',
+    versions: ['>=3'],
+  },
+  (ConnectionPromise) => {
```

Comment:
This could be moved to an external function to make the code cleaner and group hooks together.